src/util/stl-utils.h

```diff
@@ -292,6 +292,29 @@ struct CompareFirstMemberOfPair {
   }
 };
 
+/// Comparator object that does comparison on a different
```

Comment:
Ok, I removed the comparator from stl-utils.h. I will modify the sort code
to use pairs.

On Sun, Mar 6, 2016 at 11:36 PM Daniel Povey <notifications@github.com>
wrote:

> In src/util/stl-utils.h
> <https://github.com/kaldi-asr/kaldi/pull/560#discussion_r55161622>:
>
> > @@ -292,6 +292,29 @@ struct CompareFirstMemberOfPair {
> >    }
> >  };
> >
> > +/// Comparator object that does comparison on a different
>
> No you wouldn't have to write any comparator, the default one works. I'm
> saying a make vector of pairs (state-time, state-index) and sort it. you
> don't even need stable_sort, since if state_time is the same, the
> state_index gets used instead.
> … <#msg-f:1528116455082592170_>
> On Sun, Mar 6, 2016 at 11:27 PM, Vimal Manohar <notifications@github.com>
> wrote: In src/util/stl-utils.h <#560 (comment)
> <https://github.com/kaldi-asr/kaldi/pull/560#discussion_r55161297>>: > @@
> -292,6 +292,29 @@ struct CompareFirstMemberOfPair { > } > }; > > +///
> Comparator object that does comparison on a different This is where it is
> used:
> https://github.com/vimalmanohar/kaldi/blob/sequence_clean/src/nnet3/discriminative-supervision.cc#L542
> Would I have to write a similar comparator to sort the vector of pairs? —
> Reply to this email directly or view it on GitHub <
> https://github.com/kaldi-asr/kaldi/pull/560/files#r55161297>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kaldi-asr/kaldi/pull/560/files#r55161622>.
>
-- 
Vimal Manohar
PhD Student
Electrical & Computer Engineering
Johns Hopkins University